src/modules/user_exchange/user_exchange.service.ts

```diff
@@ -0,0 +1,161 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { UserExchange } from './models/user_exchange.entity';
+import { ExchangeService } from '../exchange/exchange.service';
+import { ExchangeKeyService } from '../exchangeKey/exchangeKey.service';
+import { UserService } from '../user/user.service'; // Import UserService
+import { CreateUserExchangeInput } from './dto/new-userExchange.input';
+import * as ccxt from 'ccxt';
+import { UserExchangePublicType } from './dto/userExchangePublic.type';
+import { CreateUserExchangeResponse } from './dto/createUserExchangeResponse.type';
+
+@Injectable()
+export class UserExchangeService {
+  constructor(
+    @InjectRepository(UserExchange)
+    private userExchangeRepository: Repository<UserExchange>,
+    private exchangeService: ExchangeService,
+    private exchangeKeyService: ExchangeKeyService,
+    private userService: UserService, // Inject UserService
+  ) {}
+
+  async create(
+    data: CreateUserExchangeInput,
+    userId: string,
+  ): Promise<CreateUserExchangeResponse> {
+    const { name, exchangeName, accessKey, secretKey } = data;
+
+    await this.verifyExchangeName(exchangeName);
+
+    const keyVerificationResult =
+      await this.exchangeKeyService.verifyExchangeKey(
+        exchangeName,
+        accessKey,
+        secretKey,
+      );
+    if (!keyVerificationResult) {
+      throw new Error(`Invalid API keys for exchange '${exchangeName}'.`);
+    }
+
+    // Check if the access key and secret key already exist in the database
+    const existingKey = await this.exchangeKeyService.findByKeys(
+      accessKey,
+      secretKey,
+    );
+    if (existingKey) {
+      throw new Error(`API keys already exist for another connection.`);
+    }
+
+    // Find the exchange by name to get the exchangeId
+    const exchange = await this.exchangeService.findByName(exchangeName);
+    if (!exchange) {
+      throw new NotFoundException(`Exchange '${exchangeName}' not found.`);
+    }
+
+    // Find the user by ID
+    const user = await this.userService.find(userId);
+    if (!user) {
+      throw new NotFoundException(`User with ID '${userId}' not found.`);
+    }
+
+    // if exchange Name exists and API keys are valid, then create the account connection and the exchange key
+    try {
+      const userExchangeEntity = this.userExchangeRepository.create({
+        name,
+        user,
+        exchange,
+      });
+
+      const savedUserExchange =
+        await this.userExchangeRepository.save(userExchangeEntity);
+
+      await this.exchangeKeyService.create(
+        accessKey,
+        secretKey,
+        savedUserExchange.id,
+      );
+
+      return {
+        success: true,
+        message: 'Cool! Your exchange key is added successfully!',
```

Comment:
the return message should be more subjective, since we're not using it directly to rending on client